data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarder.java

```diff
@@ -98,22 +106,20 @@ public Collection<Record<Event>> forwardRecords(final Collection<Record<Event>>
                 recordsToBeProcessedLocallyCounter.increment(entry.getValue().size());
             } else {
                 recordsToBeForwardedCounter.increment(entry.getValue().size());
-                AggregatedHttpResponse httpResponse;
                 try {
-                    httpResponse = peerForwarderClient.serializeRecordsAndSendHttpRequest(entry.getValue(),
-                            destinationIp, pluginId, pipelineName);
+                    forwardingRequestExecutor.submit(() -> {
+                        try {
+                            peerForwarderClient.serializeRecordsAndSendHttpRequest(entry.getValue(), destinationIp, pluginId,
```

Comment:
Looking at this code, we should probably split serializing from sending. It could help with the error handling. But, that is beyond the scope of this PR.